core/src/main/java/org/wildfly/extras/creaper/core/online/HttpModelControllerClient.java

```diff
@@ -157,21 +157,53 @@ public OperationResponse executeOperation(Operation operation, OperationMessageH
     public void close() throws IOException {
     }
 
+    protected ModelNode parseResponse(CloseableHttpResponse response) throws IOException {
+        ModelNode result;
+        String content = EntityUtils.toString(response.getEntity());
+        int status = response.getStatusLine().getStatusCode();
+        if (status == HttpStatus.SC_OK) {
+            result = ModelNode.fromJSONString(content);
+        } else {
+            result = new ModelNode();
+            result.get(Constants.OUTCOME).set(Constants.FAILED);
+            result.get(Constants.FAILURE_DESCRIPTION).set(content);
```

Comment:
Hey, I didn't ask for copying _everything_ from the code in WildFly Core test suite :-) I think that the original, where response was parsed for status code 500, was better.